packages/NodeNameResolver/NodeNameResolver/ClassNameResolver.php

```diff
@@ -33,7 +34,7 @@ public function getNode(): string
      */
     public function resolve(Node $node): ?string
     {
-        if (property_exists($node, 'namespacedName')) {
+        if (property_exists($node, 'namespacedName') && $node->namespacedName instanceof Name) {
```

Comment:
Here we can check for specific node types instead with instance of. That way we'll have better type control

Reply:
I updated to compare exactly with `Name::class` https://github.com/rectorphp/rector-src/pull/1359/commits/70ab37bda4d22203d07a37ccd461c85dc35d3f81

Reply:
I mean something like `$node instanceof ClassLike || $node instanceof ClassConstFetch ...`

Based on https://github.com/nikic/PHP-Parser/commit/f09f22760ef1b292c3312105532c8e84e80cce19

Reply:
I updated to instanceof `Class_` https://github.com/rectorphp/rector-src/pull/1359/commits/c360d2934f800b0ffa9fa2b1e5ece9bf44ddb95c

Reply:
if no property_exists, it will cause error

```bash
 1) Rector\Symfony\Tests\Rector\New_\StringToArrayArgumentProcessRector\StringToArrayArgumentProcessRectorTest::test with data set #5 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
Undefined property: PhpParser\Node\Stmt\Class_::$namespacedName
```

https://github.com/rectorphp/rector-src/runs/4378240342?check_suite_focus=true#step:11:18

so property_exists is needed.

Reply:
I added back property_exists check https://github.com/rectorphp/rector-src/pull/1359/commits/6596b85b8a9282cd888fb8a2ad7b1a173a416003

Reply:
This is probably due to PHPStan is using old php-parser, that does not have such property. We have to wait for it.

Reply:
Oh, I'm not correct. The package tests install older php-parser version:

```bash
- Locking nikic/php-parser (v4.13.1)
- ```

Reply:
Yes, packages test uses rector dev-main, so it goes to php-parser 4.13.1 existing.